src/assets/wise5/authoringTool/components/component-selector/component-selector.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { ComponentFactory } from '../../../common/ComponentFactory';
+import { MatDialog } from '@angular/material/dialog';
+import { ComponentInfoService } from '../../../services/componentInfoService';
+import { ComponentInfoDialogComponent } from '../component-info-dialog/component-info-dialog.component';
+
+@Component({
+  selector: 'component-selector',
+  templateUrl: './component-selector.component.html',
+  styleUrls: ['./component-selector.component.scss']
+})
+export class ComponentSelectorComponent {
+  @Output() componentSelectedEvent: EventEmitter<any> = new EventEmitter<any>();
+  @Input() componentType: string;
+  private componentTypeObject: any;
+  private description: string;
+  protected label: string;
+  private previewContent: any;
```

Comment:
Does this need to be a class variable, or should it be declared only where it's used, in preview() function?
```suggestion
```